picachu/domain/data_access_layer/engine.py

```diff
@@ -38,8 +38,29 @@ def checkout(_, connection_record, connection_proxy):
             )
 
 
-app_db_engine = create_engine(
-    build_connection_string(),
-    isolation_level='READ COMMITTED',
-    pool_pre_ping=True,
-)
+class DBEngineProvider:
+    def __init__(self):
+        self.app_db_engine = create_engine(
+            f'postgresql+psycopg2://{postgres_username}:{postgres_password}@{postgres_host}/{postgres_database}',
+            isolation_level='READ COMMITTED',
+            pool_pre_ping=True,
+        )
+
+    def build_engine_string(
+            self,
+            username: str,
+            password: str,
+            host: str,
+            database: str,
+    ) -> str:
+        self.app_db_engine = f'postgresql+psycopg2://{username}:{password}@{host}/{database}'
```

Comment:
this thing is exactly a connection string, it seems wrong to assign it to the engine, maybe have a separate class variable?